app/auth/suppliers.py

```diff
@@ -1,8 +1,63 @@
-from app.models import Supplier
+from app import db
+from app.models import AuditEvent, AuditTypes, Supplier
 
 
 def get_supplier(code):
     return Supplier.query.filter(
         Supplier.code == code,
         Supplier.status != 'deleted'
     ).first()
+
+
+def valid_supplier(user):
+    if not all(hasattr(user, attr) for attr in ["role", "supplier_code"]):
+        return False
+    return False if user.role is not 'supplier' or user.supplier_code is None else True
+
+
+def update_supplier_from_json(supplier, json_data):
```

Comment:
doesn't supplier already have an update_from_json?

Reply:
Yep, it's the code you said to not use

Reply:
I think it will work OOB with just `supplier.update_from_json(json_data)`

You really want to avoid cutting/pasting this code if at all possible

Reply:
^^ this wasn't copied. will need to transform form data to use `supplier.update_from_json(json_data)` I think it takes user: { supplier: { data }}